module1/homework/lengthSort/sort.cpp

```diff
@@ -0,0 +1,21 @@
+#include "sort.hpp"
+
+#include <algorithm>
+
+struct StrCompare {
+    bool operator()(const std::string& str1, const std::string& str2) {
+        if (str1.size() == str2.size()) {
+            return str1 < str2;
+        }
+        return str1.size() < str2.size();
+    }
+};
+
+std::deque<std::string> lengthSort(std::forward_list<std::string> inputList) {
+    inputList.sort(StrCompare());
```

Comment:
Think about getting argument as const reference and first copy data to deque (e.g. in constructor with begin and end iterators) and then sort data in deque.